Userland/Services/SpiceAgent/MessageHeader.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "MessageHeader.h"
+#include <AK/Stream.h>
+#include <AK/String.h>
+
+namespace SpiceAgent {
+
+ErrorOr<MessageHeader> MessageHeader::create(Message::Type type, u32 data_size, u32 protocol_version, u64 opaque)
```

Comment:
Normally(?) we reserve  `create()` for factory functions that are allocating memory under the hood. Since this struct is a fancy POD type, we can omit the factory function and just make the constructor public. Also, this factory simply calls the constructor, which means it doesn't need to return ErrorOr anyway.

Also while I'm here, why is the order of arguments different from the constructor arguments 🤔 .

Reply:
> Normally(?) we reserve create() for factory functions that are allocating memory under the hood...

That makes sense, I'll remove it and just use the constructor

> Also while I'm here, why is the order of arguments different from the constructor arguments

That is a good question... 😅